### PR TITLE
feat: add initial commit for flipt-client-go evaluation client

### PR DESCRIPTION
**Pull request by yquansah**
This PR introduces the `flipt-client-go` for embedding a dynamic library written with the Rust core for flipt evaluations.

Completes FLI-675

### CODE REVIEW

File: sdk/client/go/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+## Flipt Client Golang
```
Comment (markphelps) [inline sdk/client/go/README.md:1]:
```suggestion
## Flipt Client Go
```

`%s/Golang/Go/g` 😄

File: sdk/client/go/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+To use this client, you can run the following command from the root of the repository:
+
+```bash
+cargo build
+```
+
+This should generate a `target/` directory in the root of this repository, which contains the dynamic linking library built for your platform. This dynamic library will contain the functionality necessary for the Golang client to make FFI calls.
+
+You can import the module that contains the evaluation client: `go.flipt.io/flipt/flipt-client-go` and build your Go project with the environment variable:
```
Comment (markphelps) [inline sdk/client/go/README.md:15]:
```suggestion
You can import the module that contains the evaluation client: `go.flipt.io/flipt/flipt-client-go` and build your Go project with the `CGO_LDFLAGS` environment variable set:
```

File: sdk/client/go/README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+CGO_LDFLAGS="-L/path/to/lib -lengine"
+```
+
+The `path/to/lib` will be the path to the dynamic library which will have the following paths depending on your platform.
+
+- **Linux**: `{FLIPT_REPO_ROOT}/target/debug/libengine.so`
+- **Windows**: `{FLIPT_REPO_ROOT}/target/debug/libengine.dll`
+- **MacOS**: `{FLIPT_REPO_ROOT}/target/debug/libengine.dylib`
+
+A sample code will look something like the following:
```
Comment (markphelps) [inline sdk/client/go/README.md:27]:
```suggestion
You can then use the client like so:
```

**Review by markphelps — COMMENTED** [review]
couple suggestions! otherwise lgtm! 

larger convo, do we want to think about scoping the client to a single namespace instead of a slice? i forget where we landed on this?

**Comment by yquansah** [comment]
@markphelps We could do that! I expanded it to accept a list of namespaces but will most definitely be simpler if we start with 1 namespace for now.

File: go.work [hunk 4]
```diff
@@ -1,4 +1,4 @@
-go 1.21
+go 1.21.3
```
Comment (yquansah) [inline go.work:1]: It didnt seem to work with just the minor, looks like I had to use the patch as well

**Review by markphelps — APPROVED** [review]
should we change the python sdk to also only accept a single namespace at a time? curious what @GeorgeMac thinks about one vs multi namespaces? I guess we could always keep the engine accepting multi like you have here but constrain the language specific SDKs to start (also like you have here).

One way that might be forward-looking is to make `NewClient` accept functional options, where one option would be `WithNamespace`, then we could default to the `default` namespace if none provided. This would also open up the potential to adding a `ForNamespaces` option in the future?

Maybe I'm overthinking it, will be interesting to see what other languages will look like for a single vs multiple namespaces

**Comment by yquansah** [comment]
@markphelps I guess since people can use namespaces in a variety of ways it _might_ be beneficial to accept multiple namespaces for clients, but I am for constraining it also for now.

**Pull request merged**